cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/templates/slurm_parallelcluster_queue_partition.conf

```diff
@@ -1,38 +1,89 @@
+{%- import 'common/slurm_utils.conf' as slurm_utils -%}
+
 # This file is automatically generated by pcluster
-{% set ns = namespace(has_node=false) %}
+{% set ns = namespace(has_node=false) +%}
 
-{% for compute_resource in queue_config.ComputeResources %}
-    {% set static_size = compute_resource.MinCount %}
-    {% set dynamic_size = compute_resource.MaxCount - compute_resource.MinCount %}
-    {% set instance_type = compute_resource.InstanceType%}
-    {% if static_size > 0 %}
-NodeName={{ queue_name }}-st-{{ compute_resource.Name }}-[1-{{ static_size }}] CPUs={{ compute_resource | vcpus }} RealMemory={{ compute_resource | realmemory }} State=CLOUD Feature=static,{{ instance_type }},{{ compute_resource.Name }}{% if compute_resource.Efa.Enabled %},efa{% endif %}{% if instance_type | gpus > 0 %},gpu{% if not no_gpu %} Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}{% endif %}{% endif %}
+{#- Macros: #}
+{%- macro cpus(compute_resource) -%}
+    {{ ' ' }}CPUs={{ compute_resource | vcpus }}
+{%- endmacro %}
 
-    {% endif %}
-    {% if dynamic_size > 0 %}
-NodeName={{ queue_name }}-dy-{{ compute_resource.Name }}-[1-{{ dynamic_size }}] CPUs={{ compute_resource | vcpus }} RealMemory={{ compute_resource | realmemory }} State=CLOUD Feature=dynamic,{{ instance_type }},{{ compute_resource.Name }}{% if compute_resource.Efa.Enabled %},efa{% endif %}{% if instance_type | gpus > 0 %},gpu{% if not no_gpu %} Gres=gpu:{{ instance_type | gpu_type }}:{{ instance_type | gpus }}{% endif %}{% endif %}
+{%- macro node_state(state="CLOUD") -%}
+    {{ ' ' }}State={{ state }}
+{%- endmacro %}
+
+{%- macro real_memory(compute_resource) %}
+    {%- if not compute_resource.InstanceTypeList -%}
+        {{ ' ' }}RealMemory={{ compute_resource | realmemory }}
+    {%- endif -%}
+{%- endmacro %}
+
+{%- macro feature(compute_resource, is_static) -%}
+    {{ ' ' }}Feature={{ "static" if is_static else "dynamic" }}
+    {%- if not compute_resource.InstanceTypeList -%}
```

Comment:
#TODO for single instance type in InstanceTypeList we can set the instance type

Reply:
Done([feature_property](https://github.com/aws/aws-parallelcluster-cookbook/pull/1499/files#diff-6668f338ff431ea63de7cd7dd41034a6d93cf95510f2667048c366eb12fbbbbfR25)).